model/modules/path_classifier.py

```diff
@@ -30,24 +37,17 @@ def forward(self, encoded_paths: torch.Tensor, contexts_per_label: List[int],) -
         batched_context, attention_mask = cut_encoded_contexts(encoded_paths, contexts_per_label, self._negative_value)
 
         # [batch size; classifier input size]
-        initial_state = torch.cat(
-            [ctx_batch.mean(0).unsqueeze(0) for ctx_batch in encoded_paths.split(contexts_per_label)]
-        )
-        attn_weights = self.attention(initial_state, batched_context, attention_mask)
-
-        # [batch size; 1; classifier input size]
-        context = torch.bmm(attn_weights.transpose(1, 2), batched_context)
+        attn_weights = self.attention(batched_context, attention_mask)
 
         # [batch size; classifier input size]
-        context = context.view(context.shape[0], -1)
-
-        # [batch size; 2 * classifier input size]
-        concat_input = torch.cat([initial_state, context], dim=1)
+        context = torch.sum(batched_context * attn_weights, dim=1)
 
         # [batch size; classifier input size]
-        concat = torch.tanh(self.concat_layer(concat_input))
+        concat = torch.tanh(self.concat_layer(context))
```

Comment:
As I understand, after applying attention we should pass a vector to MLP, why you specify one linear layer?